### PR TITLE
Wrap getting nonce in go

### LINKED ISSUE #242: Wrap getting nonce in go
https://github.com/api3dao/airseeker-v2/blob/3afaf311177d7c5e9271fefd948cf6631f789f78/src/update-feeds-loops/submit-transactions.ts#L85

We should wrap that in go utils. Related to that, we might want to revisit how update timestamp is persisted. See [here](https://api3workspace.slack.com/archives/C05TQPT7PNJ/p1712567895101259).

### PR DESCRIPTION
**Pull request by Siegrift**
Closes https://github.com/api3dao/airseeker-v2/issues/242

### CODE REVIEW

File: src/update-feeds-loops/submit-transactions.ts [hunk 1]
```diff
@@ -82,7 +82,12 @@ export const submitTransaction = async (
         );
 
         logger.debug('Getting nonce.');
-        const nonce = await provider.getTransactionCount(sponsorWallet.address, blockNumber);
+        const goNonce = await go(async () => provider.getTransactionCount(sponsorWallet.address, 'latest'));
```
Comment (bdrhn9) [inline src/update-feeds-loops/submit-transactions.ts]: We must use specific blockNumber to not cause any race-conditioning between different RPC providers or Airseeker-v2 instances. By doing [that](https://github.com/api3dao/airseeker-v2/blob/aae424814df3cd44d5aff99212ea1fa4fc0edd82/src/update-feeds-loops/update-feeds-loops.ts#L92), we are ensuring that whole loop is working for the same state on the chain.

**Review by bdrhn9 — APPROVED** [review]
Other than the comment which needs to be addressed, LGTM.

File: src/update-feeds-loops/submit-transactions.ts (hunk 1 above)
Comment (Siegrift) [reply src/update-feeds-loops/submit-transactions.ts]: 🤦 I have no idea how did I end up writing `latest` there.

Comment (Siegrift) [reply src/update-feeds-loops/submit-transactions.ts]: I think it was because of my carelessness and Copilot...

**Pull request merged**